src/rgw/rgw_s3_filter.cc

```diff
@@ -94,44 +110,51 @@ void rgw_s3_key_value_filter::dump(Formatter *f) const {
     return;
   }
   f->open_array_section("FilterRules");
-  for (const auto& key_value : kv) {
+  for (const auto& key_value_type : kvt) {
     f->open_object_section("");
-    ::encode_json("Name", key_value.first, f);
-    ::encode_json("Value", key_value.second, f);
+    ::encode_json("Name", key_value_type.first, f);
+    ::encode_json("Value", key_value_type.second.first, f);
+    ::encode_json("Type", key_value_type.second.second == true ? "OUT" : "IN", f);
     f->close_section();
   }
   f->close_section();
 }
 
 bool rgw_s3_key_value_filter::decode_xml(XMLObj* obj) {
-  kv.clear();
```

Comment:
Hi @yuvalif, i made a mistake here. 

Need to add `kvt.clear()`, just realised the mistake while working on the other PR. 

Will make the change and squash the change.